lib/note-revisions/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import NotePreview from '../components/note-preview';
+import TagChip from '../components/tag-chip';
+import { noteTags } from '../state/selectors';
+
+import type * as S from '../state';
+import type * as T from '../types';
+
+type OwnProps = {
+  noteId: T.EntityId;
+  note?: T.Note;
+};
+
+type StateProps = {
+  tags: T.Tag[];
+  noteId: T.EntityId | null;
+  note: T.Note | null;
+};
+
+type Props = OwnProps &
+  StateProps &
+  Pick<React.HTMLProps<HTMLDivElement>, 'aria-hidden'>;
+
+export class NoteRevisions extends Component<Props> {
+  static displayName = 'NoteRevisions';
+
+  render() {
+    const { note, noteId, tags, 'aria-hidden': ariaHidden } = this.props;
+
+    return (
+      <div aria-hidden={ariaHidden} className="note-revisions">
+        <NotePreview noteId={noteId} note={note} />
+        <div className="tags">
+          {tags.map(({ name, deleted }) => (
+            <TagChip
+              key={name}
+              tagName={name}
+              interactive={false}
+              deleted={deleted}
+            />
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps: S.MapState<StateProps, OwnProps> = (state, props) => {
+  const noteId = props.noteId ?? state.ui.openedNote;
+  const note = props.note ?? state.data.notes.get(noteId);
+  const showDeletedTags = state.ui.showDeletedTags;
+  const tags = noteTags(state, note).filter(
+    (tag) => showDeletedTags || !tag.deleted
+  );
```

Comment:
Depending on `showDeletedTags` value, the deleted tags are filtered out and not displayed.